### ISSUE TITLE
Add validation check for options passed to the `qml.qnode` decorator and to `qml.QNode`

### ISSUE THREAD

**Issue by antalszava** [PennyLaneAI/pennylane#1496]
### Feature details

It would be nice to have a validation check for keyword arguments passed to `qml.qnode` and `qml.QNode`. Options that are unrecognized should raise a warning.

Consider e.g., the following:
```python
import pennylane as qml
from pennylane import numpy as np


dev = qml.device('qulacs.simulator', wires=2)

par = np.array(0.3, requires_grad=True)

@qml.qnode(dev, grad_method='adjoint')
def circuit(x):
    qml.RY(x, wires=[0])
    return qml.expval(qml.PauliZ(0))

qml.grad(circuit)(par)
```
As the keyword for choosing the differentiation method is `diff_method` instead of `grad_method`, the above snippet would execute well, although the `'qulacs.simulator'` device doesn't support adjoint differentiation. Instead of the `diff_method='adjoint'`, we'd be using `diff_method='best'` without any warnings/errors raised, just because the keyword specified was `grad_method`. Therefore the user doesn't get informed that the computation doesn't happen as intended.

### Implementation

Likely, this will involve adding a validation check in `qml.QNode` (and perhaps in the decorator). Would consider raising a warning.

Note on the implementation approach: it would make sense to rely on standard Python argument validation. So instead of using if statements in the code, the signature could be split into known arguments and kwargs.

----

*List of tasks for completion:*

1. Add a validation check mechanism to creating qnodes with either the `qml.qnode` decorator or with `qml.QNode`. The signature of `qml.qnode` and/or `qml.QNode` should be changed such that known arguments are defined as separate parameters and there's an option to pass keyword arguments too.
2. Passing options to `qml.qnode` and `qml.QNode` that are unrecognized should raise a warning. Therefore, the example in the original description and in the *Additional information* section should raise a warning about `grad_method` being unrecognized.
4. Include unit and integration testing for the changes.

### How important would you say this feature is?

2: Somewhat important. Needed this quarter.

### Additional information

The following example should raise a warning for `qml.QNode`, as `grad_method` is an unrecognized option:
```python
import pennylane as qml
from pennylane import numpy as np


dev = qml.device('qulacs.simulator', wires=2)

par = np.array(0.3, requires_grad=True)

def circuit(x):
    qml.RY(x, wires=[0])
    return qml.expval(qml.PauliZ(0))

qnode = qml.QNode(circuit, dev, grad_method='adjoint')
qml.grad(qnode)(par)
```

**Comment by Christophe-pere** [comment]
I'm interested to try to work on this. I'm a beginner with Pennylane but, seems interesting.

**Comment by antalszava** [comment]
Hi @Christophe-pere that's great to hear! :) Feel free to open a PR and ask any questions that may arise.

**Comment by ianmclean2011** [comment]
@antalszava I think my PR resolved this issue, so you might be able to close it now.

**Comment by thisac** [comment]
Thanks @ianmclean2011. Closed by #1591.
